src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultString = '';
+    let mainSeparatorClone = separator;
+
+    if (separator === null) {
+      mainSeparatorClone = 'null';
+    }
```

Comment:
Try to use this reference to remove this `if` statement



```js
console.log(String(null)) // null
console.log(typeof String(null)) // string

const test = null 

console.log(`${test}`) // null
console.log("result: " + test) // result: null
```

Feel free to write me in slack if you have any problems

[5-ways-to-convert-a-value-to-string-in-javascript](https://medium.com/dailyjs/5-ways-to-convert-a-value-to-string-in-javascript-6b334b2fc778)